### ISSUE TITLE
Allow users to control "suggestions_url" & "suggestions_url_post_params" for any custom search engine

### ISSUE THREAD

**Issue by tomahl** [brave/brave-browser#6566]
## Description 
As a user I would like to be able to set custom or choose predefined parameters for any custom search engine so that I can use a prediction service to help complete searches and URLs typed in the address bar.

When using e.g. the Vivaldi Browser users can enter additional parameters (**suggestions_url** and **suggestions_url_post_params**) for their custom search provider. As of today this doesn't exsist in the Brave UI and therefore autocomplete doesn't work.

## Steps to Reproduce

   1. Go to chrome://settings/searchEngines
   2. Click "Add"
   3. Provide information for "Search engine", "keyword" and "URL with %s in place of query"

## Expected result:
I would expect two input methods for either providing the necessary information or be able to choose from predefined values.

## Reproduces how often: 
Always

## Brave version (brave://version info)
0.69.135

## Version/Channel Information:

- Can you reproduce this issue with the current release?
- Can you reproduce this issue with the beta channel?
- Can you reproduce this issue with the dev channel?
- Can you reproduce this issue with the nightly channel?

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by srirambv** [comment]
You can add custom search engine using `brave://settings/searchEngines`
![image](https://user-images.githubusercontent.com/17010094/67205037-d7c6de80-f3dc-11e9-90e3-0bc2efd98d46.png)

**Comment by tomahl** [comment]
Yes you can, but not custom parameters like **suggestions_url** and **suggestions_url_post_params**.

**Comment by srirambv** [comment]
Could you provide more info on how this is helpful or what additional value it adds?

**Comment by tomahl** [comment]
Sure thing. If I add e.g. DuckDuckGo with my own custom URL then the prediction service that should help me complete searches and URLs typed in the address bar doesn't work. It only works if you go with the predefined version of DDD in Brave.

**Comment by shamrin** [comment]
This feature is crucial to fully try out [Cliqz](https://beta.cliqz.com/) and its search suggestions in Brave.

(Lot's of technical details about Cliqz are [at their blog](https://www.0x65.dev/).)

**Comment by sant0s12** [comment]
Any updates on this? I would also love to have this feature to have autocomplete on search engines like searx.

**Comment by sant0s12** [comment]
For anyone who is still interested, I found [this](https://stackoverflow.com/questions/35237414/add-suggestion-url-for-custom-search-engine-in-chrome) StackOverflow question, which helped me have autocomplete on Brave with searx.
Instead of adding the engine manually, just visit the page and it will be added to the search engines list. Interestingly, that will enable autocomplete once you set it as default. Being able to change `suggestions_url` and `suggestions_url_post_params` manually would still be nice though.

**Comment by bsclifton** [comment]
This issue hasn't had an update in a while - it's met the criteria for being stale. The issue is going to be closed for now. If this is still a problem, please leave a comment or re-open the issue. Thanks!

**Comment by JacksonBarker** [comment]
[#47018](https://github.com/brave/brave-browser/issues/47018) is a duplicate of this issue. I hope that eventually this issue will be re-opened and addressed, as currently, there is no way to modify `suggestions_url` on Android because the Preferences file is not exposed.
